yt_dlp/extractor/vrt.py

```diff
@@ -229,102 +319,104 @@ class VrtNUIE(VRTBaseIE):
+
+    # Perform some requests with automatic redirection disabled
+    # to be able to grab necessary info in intermediate step
+    _opener = urllib.request.build_opener(NoRedirect, urllib.request.HTTPCookieProcessor(_cookies))
 
     def _perform_login(self, username, password):
-        auth_info = self._gigya_login({
-            'APIKey': '3_0Z2HujMtiWq_pkAjgnS2Md2E11a1AwZjYiBETtwNE-EoEHDINgtnvcAOpNgmrVGy',
-            'targetEnv': 'jssdk',
-            'loginID': username,
-            'password': password,
-            'authMode': 'cookie',
-        })
 
-        if auth_info.get('errorDetails'):
-            raise ExtractorError(f'Unable to login. VrtNU said: {auth_info["errorDetails"]}', expected=True)
+        # 1. Obtain session cookies
+        # 1.a Visit 'login' URL. Get 'authorize' location and 'oidcstate' cookie
+        res = self._opener.open('https://www.vrt.be/vrtnu/sso/login', None)
+        self._auth_url = res.headers.get_all('Location')[0]
 
-        # Sometimes authentication fails for no good reason, retry
-        for retry in self.RetryManager():
-            if retry.attempt > 1:
-                self._sleep(1, None)
-            try:
-                self._request_webpage(
-                    'https://token.vrt.be/vrtnuinitlogin', None, note='Requesting XSRF Token',
-                    errnote='Could not get XSRF Token', query={
-                        'provider': 'site',
-                        'destination': 'https://www.vrt.be/vrtnu/',
-                    })
-                self._request_webpage(
-                    'https://login.vrt.be/perform_login', None,
-                    note='Performing login', errnote='Login failed',
-                    query={'client_id': 'vrtnu-site'}, data=urlencode_postdata({
-                        'UID': auth_info['UID'],
-                        'UIDSignature': auth_info['UIDSignature'],
-                        'signatureTimestamp': auth_info['signatureTimestamp'],
-                        '_csrf': self._get_cookies('https://login.vrt.be').get('OIDCXSRF').value,
-                    }))
-            except ExtractorError as e:
-                if isinstance(e.cause, HTTPError) and e.cause.status == 401:
-                    retry.error = e
-                    continue
-                raise
+        # 1.b Follow redirection: visit 'authorize' URL. Get OIDCXSRF & SESSION cookies
+        res = self._opener.open(self._auth_url, None)
+
+        # 2. Perform login
+        headers = {
+            'Content-Type': 'application/json',
+            'Oidcxsrf': self._cookies["OIDCXSRF"],
+            'Cookie': self._cookies.header(['OIDCXSRF', 'SESSION'])
+        }
+        post_data = {"loginID": f"{username}", "password": f"{password}", "clientId": "vrtnu-site"}
+        res = self._request_webpage('https://login.vrt.be/perform_login', None, note='Performing login', errnote='Login failed', fatal=True, data=json.dumps(post_data).encode(), headers=headers)
```

Comment:
I don't see why the addition of a new CookieJar and request opener is required. Can't you just do something like this?
```python
login_page = self._request_webpage('https://www.vrt.be/vrtnu/sso/login', None)
self._auth_url = login_page.url
res = self._download_json(
    'https://login.vrt.be/perform_login', None, data=json.dumps({
        "loginID": username,
        "password": password,
        "clientId": "vrtnu-site"
        }).encode(), headers={
    'Content-Type': 'application/json',
    'Oidcxsrf': self._get_cookies('https://login.vrt.be').get('OIDCXSRF').value,
})
```
If it turns out that disabling redirects is required, this should be added on the core level and not the extractor level.

Reply:
Thanks! With your proposed code snippet as an example I was able to ditch my idiosyncratic approach.

- Redirects are no longer disabled
- Only methods from extractor/common.py are used now instead of the underlying urllib
- The custom cookie handling is gone